src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -381,8 +388,25 @@ private async Task RefreshIndividualKeyValues()
                                 break;
                             }
 
-                            hasChanged = true;
-                            ProcessChanges(Enumerable.Repeat(keyValueChange, 1));
+                            if (keyValueChange.ChangeType == KeyValueChangeType.Deleted)
+                            {
+                                _watchedSettingsCache.TryRemove(watchedKeyLabel, out ConfigurationSetting watchedSeting);

```

Comment:
>watchedSeting [](start = 106, length = 13)

`watchedSeting` -> Typo? We could also use a discard `_` since we aren't using this value.